### PR TITLE
added users-controller and users-router

### PR DESCRIPTION
**Pull request by DeepthiDukka**
# Description

Please provide a short summary explaining what this PR is about.

Fixes # (issue)

# How to test?

Please provide a short summary how your changes can be tested?

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by bytewiz — CHANGES_REQUESTED** [review]
Please fix the lint error 👍  So the build doesn't break :)

File: src/server/api/controllers/messages-controller.js [hunk 1]
```diff
@@ -14,7 +14,9 @@ const getMessageById = async (id) => {
   try {
     const messages = await knex('messages')
       .select('messages.id as id', 'title')
-      .where({ id });
+      .where({
```
Comment (tpetrina) [inline src/server/api/controllers/messages-controller.js:17]: These changes feel like a formatting error, can you format this file again or remove it from PR?

File: src/server/api/controllers/modules.controller.js [hunk 2]
```diff
@@ -17,7 +17,9 @@ const getModuleById = async (id) => {
   try {
     const modules = await knex('modules')
       .select('modules.id as id', 'title')
-      .where({ id });
+      .where({
```
Comment (tpetrina) [inline src/server/api/controllers/modules.controller.js:20]: Same as before, this change is not something you want to commit.

File: src/server/api/lib/utils/pagination.js [hunk 3]
```diff
@@ -9,10 +9,7 @@
  *  order: Sort either ascending "asc" or descending "desc"
  */
 const paginate = (knex, { limit = 50, offset = 0, orderBy, order = 'asc' }) => {
-  return knex
-    .limit(limit)
-    .offset(offset)
-    .orderBy(orderBy, order);
+  return knex.limit(limit).offset(offset).orderBy(orderBy, order);
```
Comment (tpetrina) [inline src/server/api/lib/utils/pagination.js:12]: Formatting again.

File: src/server/api/lib/utils/winston.js [hunk 4]
```diff
@@ -40,9 +40,7 @@ const appendTimestamp = format((info, opts) => {
   if (opts.tz) {
     return {
       ...info,
-      timestamp: moment()
-        .tz(opts.tz)
-        .format(),
+      timestamp: moment().tz(opts.tz).format(),
```
Comment (tpetrina) [inline src/server/api/lib/utils/winston.js:43]: Formatting, you probably didn't mean this.

File: src/server/api/routes/messages-router.js [hunk 5]
```diff
@@ -98,10 +98,7 @@ router.post('/', (req, res) => {
     .catch((error) => {
       console.log(error);
 
-      res
-        .status(400)
-        .send('Bad request')
-        .end();
+      res.status(400).send('Bad request').end();
```
Comment (tpetrina) [inline src/server/api/routes/messages-router.js:101]: Formatting, probably unintended.

File: src/server/api/routes/modules.router.js [hunk 6]
```diff
@@ -101,10 +101,7 @@ router.post('/', (req, res) => {
     .catch((error) => {
       console.log(error);
 
-      res
-        .status(400)
-        .send('Bad request')
-        .end();
+      res.status(400).send('Bad request').end();
```
Comment (tpetrina) [inline src/server/api/routes/modules.router.js:104]: Formatting, probably unintended.

File: src/server/api/routes/users-router.js [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+    .getUsers()
+    .then((result) => res.json(result))
+    .catch(next);
+});
+
+/**
+ * @swagger
+ * /users/{ID}:
+ *  get:
+ *    summary: Get module by ID
```
Comment (tpetrina) [inline src/server/api/routes/users-router.js:35]: This comment should mention `user`, not `module`.

File: src/server/api/routes/users-router.js [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+ *              format: date-time
+ *            classId:
+ *              type: string
+ *    responses:
+ *      201:
+ *        description: Module created
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.post('/', (req, res) => {
```
Comment (tpetrina) [inline src/server/api/routes/users-router.js:96]: Why is this here? Your ticket mentions only implementing GET and this method `usersController.createUser` doesn't even exist! And the comment mentions the word module. Please remove.

File: src/server/api/routes/users-router.js [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+ *              format: date-time
+ *            classId:
+ *              type: string
+ *    responses:
+ *      200:
+ *        description: Module was patched
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.patch('/:id', (req, res, next) => {
```
Comment (tpetrina) [inline src/server/api/routes/users-router.js:141]: Same as above, remove.

File: src/server/api/routes/users-router.js [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+ *      - in: path
+ *        name: ID
+ *        description: ID of the module to delete.
+ *    responses:
+ *      200:
+ *        description: Module deleted
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.delete('/:id', (req, res) => {
```
Comment (tpetrina) [inline src/server/api/routes/users-router.js:166]: Same as above, remove.

File: src/server/app.js [hunk 11]
```diff
@@ -55,12 +55,12 @@ app.use((err, req, res) => {
   res.sendStatus(500);
 });
 
-app.use('/api/', function(req, res) {
+app.use('/api/', function (req, res) {
```
Comment (tpetrina) [inline src/server/app.js:58]: Formatting, please undo.

File: src/server/config/db.js [hunk 12]
```diff
@@ -1,9 +1,11 @@
-require('dotenv').config({ path: '../../.env' });
+require('dotenv').config({
```
Comment (tpetrina) [inline src/server/config/db.js:1]: Probably unintended changes, please remove.

File: src/server/migrations/20200906151245_add_users_table.js [hunk 13]
```diff
@@ -1,22 +1,16 @@
-exports.up = function(knex) {
+exports.up = function (knex) {
```
Comment (tpetrina) [inline src/server/migrations/20200906151245_add_users_table.js:1]: Probably unintended changes, please remove.

File: src/server/migrations/20200906152151_add_channels_table.js [hunk 14]
```diff
@@ -1,19 +1,13 @@
-exports.up = function(knex) {
+exports.up = function (knex) {
```
Comment (tpetrina) [inline src/server/migrations/20200906152151_add_channels_table.js:1]: Probably unintended changes, please remove.

File: src/server/migrations/20200906152538_add_channel_messages_table.js [hunk 15]
```diff
@@ -1,23 +1,17 @@
-exports.up = function(knex) {
+exports.up = function (knex) {
```
Comment (tpetrina) [inline src/server/migrations/20200906152538_add_channel_messages_table.js:1]: Probably unintended changes, please remove.

File: src/server/migrations/20200906153228_add_channel_members.js [hunk 16]
```diff
@@ -1,22 +1,16 @@
-exports.up = function(knex) {
+exports.up = function (knex) {
```
Comment (tpetrina) [inline src/server/migrations/20200906153228_add_channel_members.js:1]: Probably unintended changes, please remove.

File: .vscode/settings.json [hunk 17]
```diff
@@ -1,5 +1,8 @@
 {
   "editor.defaultFormatter": "esbenp.prettier-vscode",
   "editor.formatOnSave": true,
-  "prettier.configPath": "./.prettierrc.js"
+  "prettier.configPath": "./.prettierrc.js",
+  "[javascript]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
```
Comment (tpetrina) [inline .vscode/settings.json:6]: While this is ok, are you sure you want to commit this?

**Review by tpetrina — CHANGES_REQUESTED** [review]
Please remove changes to files you didn't want to do in this PR. Also, remove extra code.

File: src/server/api/controllers/users-controller.js [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+const getUsers = async () => {
+  try {
+    const userList = await knex('users')
+      .select('*')
+      .limit(20);
+    return userList.length > 0 ? userList : 'No Users';
+  } catch (error) {
+    return error.message;
+  }
+};
```
Comment (bytewiz) [inline src/server/api/controllers/users-controller.js:12]: One thing I think could be nice to do instead of hardcoding `20` as the limit, what we could do it making a default value which can be overwritten in case we want to make the limit something else (more flexible). It would look something like this:

```javascript
const getUsers = async ({ limit = 20 }) => {
  try {
    const userList = await knex('users')
      .select('*')
      .limit(limit);
    return userList.length > 0 ? userList : 'No Users';
  } catch (error) {
    return error.message;
  }
};
```

File: src/server/api/controllers/users-controller.js [hunk 19]
```diff
@@ -0,0 +1,28 @@
+const knex = require('../../config/db');
+
+const getUsers = async () => {
+  try {
+    const userList = await knex('users')
+      .select('*')
+      .limit(20);
+    return userList.length > 0 ? userList : 'No Users';
```
Comment (bytewiz) [inline src/server/api/controllers/users-controller.js:8]: So there you either return an `array of users` or a `string`. This might cause some funky errors later.

I would suggest: 
```suggestion
    return userList;
```

As then you return an array regardless and you can expect that an empty array means "No users"

**Review by bytewiz — CHANGES_REQUESTED** [review]

File: src/server/api/controllers/messages-controller.js [hunk 20]
```diff
@@ -61,4 +61,4 @@ module.exports = {
   deleteMessage,
   createMessage,
   editMessage,
-};
\ No newline at end of file
+};
```
Comment (benna100) [inline src/server/api/controllers/messages-controller.js:64]: Deepthi, this was exactly the thing i was talking about in class! Make sure you commit stuff like this that is not relevant to your story!!

File: src/server/api/routes/messages-router.js [hunk 21]
```diff
@@ -178,4 +178,4 @@ router.delete('/:id', (req, res) => {
     .catch((error) => console.log(error));
 });
 
-module.exports = router;
\ No newline at end of file
+module.exports = router;
```
Comment (benna100) [inline src/server/api/routes/messages-router.js:181]: Again here you are committing stuff

**Review by benna100 — APPROVED** [review]

**Comment by benna100** [comment]
Deepthi you have a merge conflict you need to fix

**Review by tpetrina — APPROVED** [review]

**Review by benna100 — APPROVED** [review]
